src/StatsBase.jl

```diff
@@ -8,7 +8,11 @@ module StatsBase
     import Base: rand, rand!
     import Base.LinAlg: BlasReal, BlasFloat
     import Base.Cartesian: @nloops, @nref, @nextract
-    import Base.Collections: heapify!, heappop!, percolate_down!
+    if VERSION < v"0.6.0-dev.1950"
+        import Base.Collections: heapify!, heappop!, percolate_down!
+    else
+        import DataStructures: heapify!, heappop!, percolate_down!
+    end
```

Comment:
If we're requiring DataStructures as a dependency anyway, we can just use
```julia
using DataStructures: heapify!, heappop!, percolate_down!
```
rather than conditioning on the version since it's also available on 0.5.